Alpaca.Markets/RestClient.cs

```diff
@@ -64,16 +81,28 @@ public sealed partial class RestClient : IDisposable
         /// <param name="polygonRestApi">Polygon REST API ennpoint URL.</param>
         /// <param name="alpacaDataApi">Alpaca REST data API endpoint URL.</param>
         /// <param name="isStagingEnvironment">If <c>true</c> use staging.</param>
+        /// <param name="maxRetryAttempts">Number of times to retry an Http request, if the status code is one of the <paramref name="retryHttpStatuses"/></param>
+        /// <param name="retryHttpStatuses">Http status codes that trigger a retry, up to the <paramref name="maxRetryAttempts"/></param>
+        /// <param name="apiVersion">Version of Alpaca api to call.  Valid values are "1" or "2".</param>
         public RestClient(
             String keyId,
             String secretKey,
             Uri alpacaRestApi,
             Uri polygonRestApi,
             Uri alpacaDataApi,
-            Boolean isStagingEnvironment)
+            Boolean isStagingEnvironment,
+            Int32 maxRetryAttempts,
+            IEnumerable<Int32> retryHttpStatuses,
+            String apiVersion)
         {
             keyId = keyId ?? throw new ArgumentException(nameof(keyId));
             secretKey = secretKey ?? throw new ArgumentException(nameof(secretKey));
+            if (maxRetryAttempts < 1) throw new ArgumentException(nameof(maxRetryAttempts));
+            _alpacaApiVersion = apiVersion ?? "1";
+            if (!_supportedApiVersions.Contains(apiVersion)) throw new ArgumentException(nameof(apiVersion));
+
+            _alpacaRestApiThrottler.MaxAttempts = maxRetryAttempts;
+            _retryHttpStatuses = retryHttpStatuses ?? new Int32[] { };
```

Comment:
Question - AFAIU by default we do not treat any HTTP status as retry status, right?

Reply:
I believe the intent is to make it so that SDK users can define which statuses they want to retry on.

Reply:
Not sure I'm reading your question correctly, but will try to respond.  The original logic would retry up to 5 times on any status other than 200 - OK.  I personally had a few cases where my logic timing was off and I'd get a 404 - Not Found, and it would spin around 5 times and get 404 every time.  I would rather have 404 reported back to me immediately, and I'll handle in my own code.  However, I am not everyone, so this was designed to create flexibility in both regards.

Reply:
I'm OK with your idea - I'm just thinking about providing default list of response codes which will work for retry.

Reply:
I think I understand where you're going now.  You mentioned this previously, that 429 and 503 could be defaulted into this list, and then no special handling of them would be necessary in the RateThrottler.  I can see the attraction of that.  I do understand 429 and 503 to be unique, though, in being "wait before retry".  Other codes that the caller asks to retry, I would expect an immediate retry.  I'm not convinced either way, so I'm going to leave it the way it is, but I won't be surprised if you change it  😄